edx_solutions_projects/models.py

```diff
@@ -141,17 +142,18 @@ class WorkgroupSubmission(TimeStampedModel):
     document_mime_type = models.CharField(max_length=255)
     document_filename = models.CharField(max_length=255, blank=True, null=True)
 
+    @property
+    def document_path(self):
+        """
+        :return: the path to the document in default storage
+        """
+        return urlparse(unquote(self.document_url)).path.lstrip('/media/')
+
     def delete_file(self):
         """
         Delete uploaded file before deleting the submission.
         """
-        path = urlparse(self.document_url).path.lstrip('/media/')
-
-        try:
-            default_storage.delete(path)
-        except OSError:
-            # It doesn't exist anymore.
-            pass
```

Comment:
@viadanna, don't we want to ignore race conditions here anymore?

Reply:
There's no side-effect when trying to delete a file that does not exist.

Reply:
Resolved per conversation on chat -> S3 ignores this.